app/models/room.rb

```diff
@@ -1,3 +1,13 @@
 class Room < ApplicationRecord
   belongs_to :hotel
+  has_many :guest_rooms
+  has_many :guests, through: :guest_rooms
+
+  validates :suite, presence: true
+  validates :rate, presence: true
+
+  def guest_count
+    # why doesn't AR/SQL not like .first.guest_count?? have to use [0] to make it work? does .first wrap the object in something (like AR) but doesn't do that with [0]?? 
+    guest_rooms.select("COUNT(guest_rooms.guest_id) AS guest_count, guest_rooms.room_id").group("guest_rooms.room_id")[0].guest_count
+  end
```

Comment:
I've also run into the issue where `.first` doesn't work but `[0]` does. If you want I can look into it further and get back to you with what I find.

One thing I do want to mention is that I believe this method is probably a little over-engineered. We want to get the guest count for a specific room, we already have an association for that with `:guests`.

```ruby
def guest_count
  guests.distinct.count
end
```